spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonServerBusAutoConfiguration.java

```diff
@@ -0,0 +1,142 @@
+    @ConditionalOnMissingQualifiedBean(qualifier = "!localSegment", beanClass = CommandBus.class)
+    public AxonServerCommandBus axonServerCommandBus(AxonServerConnectionManager axonServerConnectionManager,
+                                                     AxonServerConfiguration axonServerConfiguration,
+                                                     @Qualifier("localSegment") CommandBus localSegment,
+                                                     @Qualifier("messageSerializer") Serializer messageSerializer,
+                                                     RoutingStrategy routingStrategy,
+                                                     CommandPriorityCalculator priorityCalculator,
+                                                     CommandLoadFactorProvider loadFactorProvider,
+                                                     TargetContextResolver<? super CommandMessage<?>> targetContextResolver) {
+        return AxonServerCommandBus.builder()
+                .axonServerConnectionManager(axonServerConnectionManager)
+                .configuration(axonServerConfiguration)
+                .localSegment(localSegment)
+                .serializer(messageSerializer)
+                .routingStrategy(routingStrategy)
+                .priorityCalculator(priorityCalculator)
+                .loadFactorProvider(loadFactorProvider)
+                .targetContextResolver(targetContextResolver)
+                .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(QueryBus.class)
+    public AxonServerQueryBus queryBus(AxonServerConnectionManager axonServerConnectionManager,
+                                       AxonServerConfiguration axonServerConfiguration,
+                                       AxonConfiguration axonConfiguration,
+                                       TransactionManager txManager,
+                                       @Qualifier("messageSerializer") Serializer messageSerializer,
+                                       Serializer genericSerializer,
+                                       QueryPriorityCalculator priorityCalculator,
+                                       QueryInvocationErrorHandler queryInvocationErrorHandler,
+                                       TargetContextResolver<? super QueryMessage<?, ?>> targetContextResolver) {
+        SimpleQueryBus simpleQueryBus =
+                SimpleQueryBus.builder()
+                        .messageMonitor(axonConfiguration.messageMonitor(QueryBus.class, "queryBus"))
+                        .transactionManager(txManager)
+                        .queryUpdateEmitter(axonConfiguration.getComponent(QueryUpdateEmitter.class))
+                        .errorHandler(queryInvocationErrorHandler)
+                        .build();
+        simpleQueryBus.registerHandlerInterceptor(
+                new CorrelationDataInterceptor<>(axonConfiguration.correlationDataProviders())
+        );
+
+        return AxonServerQueryBus.builder()
```

Comment:
Nit: I'd expect `.axonServerConnectionManager(axonServerConnectionManager)` to indent to `AxonServerQueryBus.`